### PR TITLE
Order Fulfillment: Add shipment tracking with custom providers

### LINKED ISSUE #1036: Shipment Tracking: Add tracking w/ custom provider
Add the ability to create a new shipment tracking record with a custom provider. Note, the initial work for adding the card and order tracking activity should already be completed in #1035 

Screen Shot 2019-04-30 at 4 56 05 PM

Selecting a **Custom Provider** should display a new "Provider name" text input box and allow the user to enter a tracking link (optional).

### PR DESCRIPTION
**Pull request by anitaa1990**
PR 3 of 4. Fixes #1036. This PR adds logic to create a new shipment tracking record with a custom provider.

### Changes:
- [x] Added a new section "Custom" to the top of the provider list.
- [x] Selecting a Custom Provider displays a new "Provider name" text input box and allow the user to enter a tracking link (optional).
- [x] Added logic to check if the tracking to be added includes a custom provider and if so, update the entered provider name and tracking link when adding a new shipment tracking.
- [x] Added logic to remember the provider name selected by the user in sharedPrefs and retrieve it when adding a new shipment tracking. The selection is retained till the user logs out of the app.
- [x] Added a new event called `order_shipment_tracking_custom_provider_selected` when custom provider is selected from the provider list.
- [x] Updated unit tests in `OrderFulfillmentPresenterTest` to test for custom providers.

### Notes
* [PR number 1](https://github.com/woocommerce/woocommerce-android/pull/1054) and [PR number 2](https://github.com/woocommerce/woocommerce-android/pull/1064)  must be merged first.

### Screenshot

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by anitaa1990** [comment]
@AmandaRiu @nbradbury merged the PR with the feature branch. Ready for review +1

File: WooCommerce/src/main/res/layout/activity_add_shipment_tracking.xml [hunk 1]
```diff
@@ -80,6 +117,42 @@
...
+                app:layout_constraintTop_toTopOf="parent"/>
+
+            <android.support.design.widget.TextInputEditText
+                android:id="@+id/addTracking_custom_provider_url"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginBottom="@dimen/margin_medium"
+                android:gravity="top"
+                android:hint="@string/order_shipment_tracking_custom_provider_url_hint"
+                android:inputType="text"
```
Comment (nbradbury) [inline WooCommerce/src/main/res/layout/activity_add_shipment_tracking.xml]: Should this use `TYPE_TEXT_VARIATION_URI`?

**Comment by nbradbury** [comment]
@anitaa1990 This looks great! I had a minor comment about `inputType` but that's not worth holding this up. I'm good to :shipit: now.

File: WooCommerce/src/main/res/layout/activity_add_shipment_tracking.xml (hunk 1 above)
Comment (anitaa1990) [reply WooCommerce/src/main/res/layout/activity_add_shipment_tracking.xml]: Thanks for the quick review @nbradbury! I've updated the fix in this [PR](https://github.com/woocommerce/woocommerce-android/pull/1078) 👍

**Pull request merged**